src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -29,6 +29,10 @@ DlgPrefWaveform::DlgPrefWaveform(QWidget* pParent, MixxxMainWindow* pMixxx,
                                       handles[i].getType());
     }
 
+    // Populate beatgrid options.
+    beatGridModeComboBox->addItem(tr("Beats"));
+    beatGridModeComboBox->addItem(tr("Beats + Downbeats"));
```

Comment:
Read the value of the setting from the config file and select the corresponding combobox item here

Reply:
Oh I see the combobox item is set in slotUpdate. However, the config value is not read on startup and  `WaveformWidgetFactory::instance()->setBeatGridMode` is not called until changed settings are applied.

Reply:
I think you can just call slotUpdate in the constructor.

Reply:
The call already exists. And I checked boot configuration, everything seems to load in place.

`mixxx.cpp` Line 448
```cpp
    WaveformWidgetFactory::instance()->setConfig(pConfig);
```

Reply:
I tested again, and yes the combobox in DlgPrefWaveform is loaded correctly. However, I cannot get downbeats to show on the waveform until I open the preferences and change the combobox value and that has to be repeated every time I start Mixxx.